src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -34,8 +36,8 @@ public TypeFactory(OutputLibrary library)
 
         public CSharpType CreateType(InputType inputType) => inputType switch
         {
-            InputLiteralType literalType       => CreateType(literalType.LiteralValueType),
-            InputUnionType unionType           => new CSharpType(typeof(object), unionType.IsNullable),
+            InputLiteralType literalType       => throw new InvalidOperationException("Literal type shouldn't be used outside of the Input layer"),
```

Comment:
For TypeFactory, we need to support `InputLiteralType`, and convert it to the right csharp type. We may not throw exception here.

Reply:
Literal is translated into corresponding type in JSON converter, similar to `CodeModelConverter`